### ISSUE TITLE
How to use the Unlit Material?

### ISSUE THREAD

**Issue by nagman** [KhronosGroup/glTF-Blender-Exporter#315]
Hi,

I'm trying to export a simple scene with some objects, a camera and no light, in order to use them in Three.js.
My meshes have textures and are shadeless.
I can't see anywhere in Blender nor in the glTF-Blender-Exporter-master folder, how can I create a simple unlit material.

### Pictures to explain:
Here's the original texture.
![floor-grand cru](https://user-images.githubusercontent.com/16116486/46581154-95ca3500-ca33-11e8-9b18-e41df1a77ed2.jpg)

I've mapped it onto a mesh.
![capture](https://user-images.githubusercontent.com/16116486/46581163-cdd17800-ca33-11e8-8c2c-58140bafac35.JPG)

I don't need any lights nor shaders as they are already baked on the texture.
So I want my mesh to be exported "as is" without any material.
I've read [here](https://github.com/KhronosGroup/glTF/issues/1215):
> Sounds good. There's still some ongoing discussion for fallback behavior, but I think we'd encourage you to `"set metal and roughness to 0 and 1 (or another high value)"` _AND_ include `KHR_materials_unlit` on the material and listed under `optionalExtensions` `extensionsUsed` — the extension is essentially just a flag to let extension-aware engines know it's shadeless, while the PBR metal/rough values are there for reasonable fallback.

But it means that every time I modify the 3D model, I need to open the .gtfl file and add those lines. Can't Blender export the `KHR_materials_unlit` automatically?

**Comment by nagman** [comment]
@donmccurdy What "Shadeless" checkbox?
Here's my material. Where can I find it?
![capture](https://user-images.githubusercontent.com/16116486/46601389-2ca6f800-caed-11e8-9766-4c1eab87e0ad.JPG)

**Comment by nagman** [comment]
Oh, I see. It's in the Blender Render engine.
But in the same scene I have PBR materials (with cycles) and shadeless materials.
So I need to have shadeless materials in a Cycles environment.
Is it possible?

**Comment by nagman** [comment]
Maybe with an emission shader? What do you think?
But it will may create heavier renders in the browser.

**Comment by donmccurdy** [comment]
Hm, I'm not sure how to export both PBR and unlit materials at the same time. Emissive will give you the same visual effect but somewhat heavier renders, as you say. What are you using to render? You may be able to set Custom Properties on the material or mesh in blender, choose `Export extras` in the export options, and then just switch the marked materials to unlit in your renderer.

**Comment by nagman** [comment]
I'm using Three.js. I don't really get what you mean by custom properties, but the procedure I have everytime I make an export is:
1. Open the .gltf with sublime text
2. Add `"KHR_materials_unlit"` to the `"extensionsUsed"` array
3. Add `"extensions": {"KHR_materials_unlit": {}}` to every unlit material

**Comment by donmccurdy** [comment]
Custom Properties (https://easyblend.org/html/data_system/custom_properties.html) can be used in Blender to add data to objects for arbitrary use. In this case you could add an "unlit=1" property to each unlit mesh, use PBR materials on them as usual, then convert at runtime. In threejs:

```js
var loader = new THREE.GLTFLoader();
loader.load( 'model.glb', function (gltf) {
  var model = gltf.scene;
  model.traverse(function (o) {
    if (o.isMesh && o.userData.unlit) {
     o.material = new THREE.MeshBasicMaterial({map: o.material.map});
    }
  });
  scene.add(model);
});
```

**Comment by nagman** [comment]
Great solution, yet the `three-gltf-loader` doesn't seem to import extras as userData.
The .gltf file (part of it):
```
{
    "extras" : {
        "unlit" : 1.0
    },
    "name" : "Material.002",
    "pbrMetallicRoughness" : {
        "baseColorTexture" : {
            "index" : 4
        },
        "metallicFactor" : 0.0
    }
},
```
But when I console.log the node in three.js, I get an empty `userData` object.

**Comment by nagman** [comment]
You made my day 😄

**Comment by spiraloid** [comment]
This is great.  Any chance that https://gltf-viewer.donmccurdy.com/  gets and update to check for the value of a custom unlit property?

**Comment by spiraloid** [comment]
![image](https://user-images.githubusercontent.com/36362743/54737096-836c2f80-4b6b-11e9-802c-54b0fb14ea47.png)

**Comment by spiraloid** [comment]
[vertex_color_bake_test.zip](https://github.com/KhronosGroup/glTF-Blender-Exporter/files/2991066/vertex_color_bake_test.zip)

**Comment by spiraloid** [comment]
[vertex_color_bake_test.zip](https://github.com/KhronosGroup/glTF-Blender-Exporter/files/2991068/vertex_color_bake_test.zip)

**Comment by donmccurdy** [comment]
This issue is out of date – see the docs for the new exporter (https://github.com/KhronosGroup/glTF-Blender-IO) included in Blender 2.8 here: https://docs.blender.org/manual/en/dev/addons/io_gltf2.html.

> To create Shadeless (Unlit) materials, use the Background material type.

My viewer only uses the official unlit material type, not `extras`.
